In [24]:

from googleapiclient import discovery
import pandas as pd

#API

API_KEY = 'AIzaSyAFSKd9h1GxOmXV-QjSSPOv1zPAWP7uOjg'


client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'friendly greetings from python' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(response)


# Hypothesis: Perspective will make more mistakes in trash talk and slang indicating it may not be toxic.
#testing
csv_file_path = '/content/comments - Sheet1.csv'
df_test_comments = pd.read_csv(csv_file_path)


#results
results = []

threshold = 0.5

for comment in test_comments:
    toxicity_score = get_toxicity_score(comment)
    label = "Toxic" if toxicity_score > threshold else "Non-Toxic"
    results.append({'Comment': comment, 'Toxicity Score': toxicity_score, 'Label': label})

#dataframe
pd.set_option('display.max_rows', None)
df_results = pd.DataFrame(results)
df_results


{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 30, 'score': {'value': 0.02543884, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.02543884, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}


Comment  Toxicity Score      Label
0                           You aint all that.        0.024496  Non-Toxic
1   It looks good on you, but I don't like it.        0.110577  Non-Toxic
2                             You're an idiot.        0.928801      Toxic
3                             you just a punk.        0.469820  Non-Toxic
4                               You not scary.        0.035220  Non-Toxic
5                               that is crazy.        0.285473  Non-Toxic
6                             I hate your hair        0.681338      Toxic
7                                Get good kid.        0.067380  Non-Toxic
8                             Love this movie.        0.018346  Non-Toxic
9                                Who... cares.        0.100457  Non-Toxic
10             You suck at this game get good.        0.767545      Toxic

I have learned about the observation and studying you can conduct with Perspective API, which would classify whether each comment is toxic or non-toxic. Through this analysis, I formed a hypothesis that states, “ Perspective will make more mistakes on trash talk and slang on thinking it may not be toxic.” With this hypothesis in mind, I noticed that it was true by using slang and gamer lingo, which the Perspective API reported were non-toxic comments. Therefore, I believe that bias may exist in the model.

From the data collected from the data frame and comments, there can be subjectivity and cultural bias. What is considered toxic or offensive can vary depending on communities, cultures, and regions. The model might not accurately captures these context, leading to biased assessments. For instance, my results for the slang and lingo I used were:

Comment: You aint all that., Toxicity Score: 0.02449606, Label: Non-Toxic
Comment: you just a punk., Toxicity Score: 0.46982017, Label: Non-Toxic
Comment: You not scary., Toxicity Score: 0.035220183, Label: Non-Toxic
Comment: Get good kid., Toxicity Score: 0.067380086, Label: Non-Toxic
Comment: Who... cares., Toxicity Score: 0.10045706, Label: Non-Toxic
Comment: You suck at this game get good., Toxicity Score: 0.7675452, Label: Toxic

The result of this could be that the Perspective API only looks into sentences that have negative words in them.

Full Result:
Comment: You aint all that., Toxicity Score: 0.02449606, Label: Non-Toxic
Comment: It looks good on you, but I don't like it., Toxicity Score: 0.11057663, Label: Non-Toxic
Comment: You're an idiot., Toxicity Score: 0.9288007, Label: Toxic
Comment: you just a punk., Toxicity Score: 0.46982017, Label: Non-Toxic
Comment: You not scary., Toxicity Score: 0.035220183, Label: Non-Toxic
Comment: that is crazy., Toxicity Score: 0.2854733, Label: Non-Toxic
Comment: I hate your hair, Toxicity Score: 0.681338, Label: Toxic
Comment: Get good kid., Toxicity Score: 0.067380086, Label: Non-Toxic
Comment: Love this movie., Toxicity Score: 0.01834645, Label: Non-Toxic
Comment: Who... cares., Toxicity Score: 0.10045706, Label: Non-Toxic
Comment: You suck at this game get good., Toxicity Score: 0.7675452, Label: Toxic
